pkg/apis/gcp/validation/backupbucket.go

```diff
@@ -0,0 +1,33 @@
+// SPDX-FileCopyrightText: 2024 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package validation
+
+import (
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	apisgcp "github.com/gardener/gardener-extension-provider-gcp/pkg/apis/gcp"
+)
+
+// ValidateBackupBucketConfig validates a BackupBucketConfig object.
+func ValidateBackupBucketConfig(config *apisgcp.BackupBucketConfig, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if config.Immutability != nil {
+		// Currently, only 'bucket' type is supported. In the future, 'object' type will be supported.
```

Comment:
No need to mention anything about `object type` in the source code, in my opinion.

Reply:
Fixed as part of c5c15f9afe942ba8a3e1c0fd658312f0b469b5be